src/main/java/org/jenkinsci/plugins/mesos/MesosApi.java

```diff
@@ -190,6 +207,13 @@ private PodSpec buildMesosAgentTask(double cpu, double mem) {
     return podSpec;
   }
 
+  private PodSpec getKillSpec(String podId) {
+    PodId id = new PodId(podId);
+    PodSpec spec = specMap.get(id);
+    // set goal to null to trigger a kill of this task
+    return new PodSpec(spec.id(), null, spec.runSpec());
+  }
```

Comment:
I would move this into something like a `MesosPodSpecTemplate` as the Nomad plugin does.

Reply:
looks like you are adding that in a seperate PR

Reply:
Yep. The easiest would be for my pull request to merge first.